cmake/pcl_options.cmake

```diff
@@ -64,3 +64,9 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 option(BUILD_tools "Useful PCL-based command line tools" ON)
 
 option(WITH_DOCS "Build doxygen documentation" OFF)
+
+# set index size
+set(PCL_SIZE_INDEX 32 CACHE STRING "Set index size. Available options are: 8 16 32 64")
+
+#set whether indices are signed or unsigned
+set(PCL_SIGNED_INDEX true CACHE STRING "Set whether indices need to be signed or unsigned. Signed by default.")
```

Comment:
If I directly change this to a `BOOL`, 
```cmake
set(PCL_INDEX_SIGNED true CACHE BOOL "Set whether indices need to be signed or unsigned. Signed by default.")
```
then the config file would change to 
```c++
#define PCL_INDEX_SIGNED ON
```
which is not the desired result. I can also accept a bool in the cmake configuration and internally turn it into a cmake string with `true` or `false`. This would provide the user with a boolean option and internally achieve the desired result, but that seems like a very messy solution.

As far as I understand, the best option would be to change the config to
```c++
#cmakedefine PCL_INDEX_SIGNED true
#ifndef PCL_INDEX_SIGNED
#define PCL_INDEX_SIGNED false
#endif
```
which achieves the desired effect. I'm not sure if there's a simpler solution that I'm missing.

Reply:
> I can also accept a bool in the cmake configuration and internally turn it into a cmake string with `true` or `false`. This would provide the user with a boolean option and internally achieve the desired result, but that seems like a very messy solution.

My belief is that CMake users will expect the option to parse properly all accepted forms of TRUE and FALSE i.e.
> True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number. False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND. Named boolean constants are case-insensitive. If the argument is not one of these specific constants, it is treated as a variable or string and the following signature is used.

Hence you internally should rely on cmake's `if` parsing to produce the correct end result `true`/`false` string representation.